pkg/prometheus/resource_selector.go

```diff
@@ -125,7 +125,7 @@ func (rs *ResourceSelector) SelectServiceMonitors(ctx context.Context, listFn Li
 
 			smKey := fmt.Sprintf("serviceMonitor/%s/%s/%d", sm.GetNamespace(), sm.GetName(), i)
 
-			if err = rs.store.AddBearerToken(ctx, sm.GetNamespace(), endpoint.BearerTokenSecret, smKey); err != nil {
+			if err = rs.store.AddBearerToken(ctx, sm.GetNamespace(), *endpoint.BearerTokenSecret, smKey); err != nil {
```

Comment:
I'd rather update the `AddBearerToken()` method to check for nil parameter and return early.

Reply:
Hi @simonpasquier changed, if you could review!